### PR TITLE
Document multipart upload for proxied artifact access

### LINKED ISSUE #12776: [DOC-FIX] MLFLOW_ENABLE_PROXY_MULTIPART_UPLOAD docs were (accidentally?) removed
### Willingness to contribute

No. I cannot contribute a documentation fix at this time.

### URL(s) with the issue

https://www.mlflow.org/docs/latest/tracking.html

### Description of proposal (what needs changing)

As noted in https://github.com/mlflow/mlflow/pull/10471#issuecomment-2248633266, that PR removed a bunch of really important details about `MLFLOW_ENABLE_PROXY_MULTIPART_UPLOAD` -- in both how to use it and how it works. I think it's important to call out because the server still maintains credentials for reading / writing the files, but clients actually perform the uploads. In this way, it can actually be a preferred approach when it's desirable to not give clients their own credentials, but to still allow their file uploads to scale horizontally rather than making the artifact server a performance bottleneck.

These docs were originally part of https://github.com/mlflow/mlflow/commit/d0062b2dbac7e4eb5013040c31ce1202dfb71865#diff-6eadaad38a8167d8665725f9405be4c742984310009aabaa11452d3dd1a11ef4R1204-R1237 which was merged just prior to the tracking server doc overhaul, so I suspect they were removed by accident.

It would also be good to document `MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE` and `MLFLOW_MULTIPART_UPLOAD_MINIMUM_FILE_SIZE ` as they are relevant to how the client behaves.

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/12786?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12786/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12786
```

</p>
</details>

### Related Issues/PRs

Resolve #12776

### What changes are proposed in this pull request?

https://github.com/mlflow/mlflow/commit/d0062b2dbac7e4eb5013040c31ce1202dfb71865 added docs for multipart upload for proxied artifact access, but it was accidentally removed in https://github.com/mlflow/mlflow/pull/10471. This PR restores the docs.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Review by serena-ruan — APPROVED** [review]
LGTM!

**Review by B-Step62 — APPROVED** [review]
LGTM! Maybe we can mark this feature stable now.

**Comment by harupy** [comment]
> Maybe we can mark this feature stable now.

sounds good!

File: docs/source/tracking/artifacts-stores.rst [hunk 1]
```diff
@@ -210,3 +210,36 @@ Deletion Behavior
 In order to allow MLflow Runs to be restored, Run metadata and artifacts are not automatically removed
 from the backend store or artifact store when a Run is deleted. The :ref:`mlflow gc <cli>` CLI is provided
 for permanently removing Run metadata and artifacts for deleted runs.
+
+Multipart upload for proxied artifact access
+============================================
+
+.. note::
+    This feature is experimental and may be changed or removed in a future release without notice.
+
```
Comment (harupy) [inline docs/source/tracking/artifacts-stores.rst:219]:
```suggestion
```

**Pull request merged**